docs/en/deploy/Running-Alluxio-On-Docker.md

```diff
@@ -106,18 +106,17 @@ contents with
 $ docker exec ${container_id} cat /opt/alluxio/conf/alluxio-site.properties
 ```
 
-### Run in High-Availability Mode
+### Run in High-Availability Mode with Zookeeper
 
 A lone Alluxio master is a single point of failure. To guard against this, a production
-cluster should run multiple Alluxio masters and use Zookeeper for leader election. One
-of the masters will be elected leader and serve client requests. If it dies, one of the
-remaining masters will become leader and pick up where the previous master left off.
+cluster should run multiple Alluxio masters and use Embedded Journal or Zookeeper 
```

Comment:
Is it possible to use Zookeeper and the embedded journal together? Genuine question because I don't have a clue.

Reply:
Well according to the docs it is possible, but doesn't describe exactly how to it

https://www.alluxio.org/docs/1.8/en/Getting-Started.html#embedded-journal-configuration

Reading the "Optional Configuration" section tells me the master rpc addresses PK is ignored if zookeeper is enabled, but doesn't say what other properties should be configured if we want to use the embedded journal and zookeeper

Reply:
We have two HA mode:
(1) Embedded journal (will be default soon)
(2) Zookeeper

These two cannot work together. If we set the `alluxio.master.journal.type=EMBEDDED` and set the Zookeeper configurations, we cannot start AlluxioMaster because `RaftJournalSystem` will complain.

The reason is that embedded journal relies on Copycat (the raft protocol you mentioned before). which has built-in leader election. The way the embedded journal works, the Copycat leader must match the Alluxio leader. So we cannot use Zookeeper, because then we will have two leaders which might not match. 

I will mention the reason in the docs later. Users may feel confusing about the embedded journal and thought it is only a journal thing, but actually, it is also a leader election thing.

Reply:
See my comment here: https://github.com/Alluxio/alluxio/pull/8486#discussion_r260620489

Basically I think we should use the term "Embedded Journal" sparingly because it doesn't describe exactly what's Alluxio is going to do. In reality it's just the name of the feature (and the journal itself, when properly configured). IMO we should stray away from using the blanket term "Embedded Journal" when referring to the non-Zookeeper HA mechanisms that are used when the feature is configured.

Others may have a different opinion from me though

Reply:
We previously call it `Raft Journal`, but then I guess maybe they think `Embedded Journal` can describe the circumstances better so they change it to `Embedded Journal`. I agree that `Embedded Journal` does not describe exactly what the feature does, but it's hard to find out an alternative name that can tell its embedded journal ability and high availability ability.

@aaudiber @calvinjia What user-facing name and description do you feel is better for the current embedded journal feature?

Reply:
I think `Embedded Journal` is a good name for describing the journaling aspect of the features. It so happens that it also handles consensus (ie. for leader election), which I think should have another term. `Internal Leader Election` or `Internal Quorum`?

Reply:
+1 for internal leader election. We can say things like "To avoid having the master as a single point of failure, Alluxio uses either Zookeeper-based leader election or internal leader election... To use internal leader election, you must configure Alluxio to use the embedded journal."